go/marshal/decode.go

```diff
@@ -305,6 +337,17 @@ func nomsValueDecoder(v types.Value, rv reflect.Value) {
 	rv.Set(reflect.ValueOf(v))
 }
 
+func marshalerDecoder(t reflect.Type) decoderFunc {
+	return func(v types.Value, rv reflect.Value) {
```

Comment:
Does this need to go in decoderCache? For that matter, should marshalerEncoder have used the encoderCache?